pkg/cache/v3/linear.go

```diff
@@ -131,6 +153,34 @@ func (cache *LinearCache) notifyAll(modified map[string]struct{}) {
 		cache.respond(value, nil)
 	}
 	cache.watchAll = make(watches)
+
+	cache.updateVersionMap(modified)
+
+	for id, watch := range cache.deltaWatches {
+		res := cache.respondDelta(watch.Request, watch.Response, watch.StreamState)
+		if res != nil {
+			delete(cache.deltaWatches, id)
+		}
+	}
+}
+
+func (cache *LinearCache) respondDelta(request *DeltaRequest, value chan DeltaResponse, state stream.StreamState) *RawDeltaResponse {
+	resp := createDeltaResponse(context.Background(), request, state, resourceContainer{
+		resourceMap:   cache.resources,
+		versionMap:    cache.versionMap,
+		systemVersion: cache.getVersion(),
+	})
+
+	// Only send a response if there were changes
+	if len(resp.Resources) > 0 || len(resp.RemovedResources) > 0 {
+		if cache.log != nil {
+			cache.log.Debugf("node: %s, sending delta response with resources: %v removed resources %v wildcard: %t",
```

Comment:
Maybe we should denote this log statements are coming from a linear cache? They look really similar to the simple cache statements

Reply:
I added `[linear cache]` prefix to all logging statements in the linear cache, so we no longer have identical messages from different components. I didn't touch any messages from `simple.go` (I don't want to modify only delta related log statements because I think we should add tags to log messages consistently). 

I believe this problem could be fixed in a generic ways similar to how all popular logging frameworks do that, here is one sample: https://pkg.go.dev/go.uber.org/zap#Logger.With We can use a different logger (which is a breaking change) or create custom functionality to support child loggers with custom tags, which are attached to every message automatically, without changing logger interface. Anyway, I think this is out of scope for this PR.

Reply:
Ah awesome thank you. I have a PR up that I could probably introduce that into. It's probably worth adding tags to logs